src/containers/Nodes/columns/constants.ts

```diff
@@ -16,3 +20,42 @@ export const DEFAULT_NODES_COLUMNS: NodesColumnId[] = [
 ];
 
 export const REQUIRED_NODES_COLUMNS: NodesColumnId[] = ['NodeId'];
+
+const ALL_NODES_GROUP_BY_PARAMS = [
+    'SystemState',
+    'Host',
+    'DC',
+    'Rack',
+    'Database',
+    'Version',
+    'Uptime',
+    'Missing',
+    'DiskSpaceUsage',
+] as const satisfies NodesGroupByField[];
+
+function getAvailableNodesGroupByParams(withSystemStateGroupBy?: boolean) {
+    if (!withSystemStateGroupBy) {
+        return ALL_NODES_GROUP_BY_PARAMS.filter((param) => param !== 'SystemState');
+    }
+
+    return ALL_NODES_GROUP_BY_PARAMS;
+}
+
+export function getNodesGroupByOptions(withSystemStateGroupBy?: boolean): SelectOption[] {
+    return getAvailableNodesGroupByParams(withSystemStateGroupBy).map((param) => {
+        return {
+            value: param,
+            content: NODES_COLUMNS_TITLES[param],
+        };
+    });
+}
+
+export function parseNodesGroupByParam(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    param?: any,
```

Comment:
Is it possible to set `string` type?

Reply:
if `Array.some` will be used instead of `includes`, no problems with type will appear

Reply:
`Array.some` doesn't set type to the passed arg. I found workaround with `Array.find`